frontend/src/metabase/query_builder/components/DataSelector.jsx

```diff
@@ -773,9 +797,14 @@ export class UnconnectedDataSelector extends Component {
       return;
     }
 
-    if (this.props.setDatabaseFn) {
+    // callback only if selected DB id is new
+    if (
+      this.props.setDatabaseFn &&
+      this.props.selectedDatabaseId !== database?.id
```

Comment:
`setDatabaseFn` callback is call only if the `selectedDatabaseId` has changed and it will also avoid the callback method to be called on component creation.